base/complex.jl

```diff
@@ -897,6 +897,17 @@ function complex(A::AbstractArray{T}) where T
     convert(AbstractArray{typeof(complex(zero(T)))}, A)
 end
 
-## promotion to complex ##
+## Promotion to complex ##
 
 _default_type(T::Type{Complex}) = Complex{Int}
+
+## Machine epsilon for complex ##
+
+function eps(z::Complex{T}) where {T<:AbstractFloat}
+    x, y = eps(real(z)), eps(imag(z))
+    if x < y
```

Comment:
Maybe a comment that this is just `hypot(x,y)` and why you aren't calling `hypot`.

Reply:
And documentation?

Reply:
This was already added to the eps docstring.

Reply:
I don't understand why you have to inline `hypot` even if it is defined later.   Julia allows you to define functions "out of order" … all that matters is that `hypot` is defined by the time `eps(::Complex)` is actually called.

Reply:
@musm mentioned that's what the test failure was, so I gave it a try. I am not entirely sure what's going on with the tests since if I throw this into the REPL it works just fine.